### PR TITLE
BAU: running integration tests

### PR DESCRIPTION
**Pull request by jkunle**
## Proposed changes

update instructions on running integration-tests locally

**Fix issues with jest config for integration tests**

- Use the latest version of jest for integration tests (ignore the config warnings - this is a bug to be fixed in an upcoming version)
- Correctly set test matching regex
- Rename jest config files so they can be picked up automatically if running in the `aws` or `mocked` directories (the same pattern as for the lambdas)
- Increase the timeout on the NINO check happy test so the tests can be run in parallel as the test suites don't interact with each other

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,11 +20,16 @@ To run these tests the following environment variables are needed:
 Temporary credentials can be found by going to the [AWS start page](https://uk-digital-identity.awsapps.com/start#/), selecting the account and clicking
 "Command line or programmatic access"
 
-To run the tests: `npm run test:aws`
+To run the tests on aws: cd into `integration-tests` folder
+
+`npm run test:aws -- --config tests/aws/jest.config.aws.ts`
```
Comment (CharlesIC) [inline README.md:25]:
```suggestion
To run the tests against a stack deployed in AWS, authenticate to the correct account and run

`STACK_NAME=<stack-name> npm run test:aws --workspace integration-tests`

_You can omit the `--workspace` parameter if the current working directory is `integration-tests`._
```

File: README.md [hunk 2]
```diff
@@ -20,11 +20,16 @@ To run these tests the following environment variables are needed:
...
 
 ### Mocked
 
-These tests run locally on your system. To run: `npm run test:mocked`
+These tests run locally on your system.
+To run mocked step function tests: cd into `integration-tests` folder
+
+`npm run test:mocked -- --config tests/aws/jest.config.mocked.ts`
```
Comment (CharlesIC) [inline README.md:32]:
```suggestion
To run the mocked step function tests run
`npm run test:mocked --workspace integration-tests`

_You can omit the `--workspace` parameter if the current working directory is `integration-tests`._
```

File: README.md [hunk 3]
```diff
@@ -20,11 +20,16 @@ To run these tests the following environment variables are needed:
...
 
-To run the tests: `npm run test:aws`
+To run the tests on aws: cd into `integration-tests` folder
+
+`npm run test:aws -- --config tests/aws/jest.config.aws.ts`
 
 ### Mocked
 
-These tests run locally on your system. To run: `npm run test:mocked`
+These tests run locally on your system.
```
Comment (CharlesIC) [inline README.md:29]:
```suggestion
These tests run locally on your system and need the Docker agent to be running in the background before their execution begins.
```

File: integration-tests/tests/mocked/nino-check/nino-check-happy.test.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { HistoryEvent } from "@aws-sdk/client-sfn";
 import { SfnContainerHelper } from "./sfn-container-helper";
 
-jest.setTimeout(30_000);
```
Comment (merlinc) [inline integration-tests/tests/mocked/nino-check/nino-check-happy.test.ts:4]: I'm assuming this delay is due to waiting for containers to startup.

Given a choice, I don't like increasing these timeouts, as it hides slow tests, and has a tendency to dramatically increase test runs when tests fail.

Is there any sort of healthcheck that can be monitored and used in https://jestjs.io/docs/configuration#globalsetup-string instead?

Comment (jkunle) [reply integration-tests/tests/mocked/nino-check/nino-check-happy.test.ts:4]: I suppose we can create a ticket to investigate this https://govukverify.atlassian.net/browse/OJ-2188

Comment (CharlesIC) [reply integration-tests/tests/mocked/nino-check/nino-check-happy.test.ts:4]: This is a good shoutout @merlinc but out of scope for this PR - agree we should haev a ticket to spike a solution and apply to all the relevant issues

**Review by surajkumar — APPROVED** [review]

**Pull request merged**